app/jobs/set_governing_apo_job.rb

```diff
@@ -57,4 +62,22 @@ def update_druid_count
     bulk_action.update(druid_count_total: pids.length)
     bulk_action.save
   end
+
+  def open_new_version(object, log)
+    if DorObjectWorkflowStatus.new(object.pid).can_open_version?
+      begin
+        vers_md_upd_info = {
+          :significance => 'minor',
+          :description => 'Set new governing APO',
+          :opening_user_name => current_user.to_s
+        }
+        object.open_new_version({:vers_md_upd_info => vers_md_upd_info})
+      rescue Dor::Exception => e
+        log.puts("#{Time.current} Failed to open new version for #{object.pid} (bulk_action.id=#{bulk_action.id}): #{e}")
+        return
```

Comment:
should this increment `:druid_count_fail`?  if so, i'd advocate for just letting the exception propagate up and letting the `rescue` that catches all errors for `set_governing_apo_and_index_safely` deal with it, and then i'd just get rid of the wrapping `rescue` here.  (and i'd prob use that to bolster my case for returning to the `check_can_set_governing_apo!` and just dropping its first check about `obj.allows_modification?`, since that would be handled by the newly added `open_new_version(current_obj, log) unless current_obj.allows_modification?` just below the check.